classes/Profile.php

```diff
@@ -214,7 +214,7 @@ private static function fillCacheAccesses($idProfile, $type, $cacheData = [])
     private static function findIdTabByAuthSlug($authSlug)
     {
         preg_match(
-            '/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z]+)_(?P<auth>[A-Z]+)/',
+            '/ROLE_MOD_[A-Z]+_(?P<classname>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)_(?P<auth>[A-Z]+)/',
```

Comment:
I'm not sure about the `\x7f-\xff` part.

While theoretically speaking "¼¸¬" would be a valid className, I'm not sure if it's a good idea to allow those characters there.

In addition, why did you change the `+` into `*`? That allow `ROLE_MOD_FOO__BAR`.

Reply:
I'm not sure about the \x7f-\xff part, mmm php accept it,  http://php.net/manual/en/language.oop5.basic.php , Œ is part of this range 

about + into * ? , it is not really change the regular expression has now tow parts, first car = [a-zA-Z_\x7f-\xff] (a valid class name can't start whit a number) and, rest of classname = [a-zA-Z0-9_\x7f-\xff]* , so at minimum one car is required

after reading again the expression, i constat that i forgot that a role must be Uppercase,
`/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z][A-Z0-9]*)_(?P<auth>[A-Z]+)/`
may be the best choice here and cover all the possibilities exept  \x7f-\xff and underscores inside class names

Reply:
> I'm not sure about the \x7f-\xff part, mmm php accept it, http://php.net/manual/en/language.oop5.basic.php , Œ is part of this range

Yeah, that's why I spoke about "theoretically valid class names". What was actually troubling me was the possible unforeseen side effects of allowing non-word characters in the roles.

You're right about the `*` though, I got mixed up when reading the regex 👍